android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/Utils.kt

```diff
@@ -11,80 +11,65 @@ private const val SEARCH_TYPE_SAP_FOLLOW_ON = "sap-follow-on"
 private const val SEARCH_TYPE_SAP = "sap"
 private const val SEARCH_TYPE_ORGANIC = "organic"
 private const val CHANNEL_KEY = "channel"
-private val validCodeSet = setOf(
-    "MOZ2", "MOZ4", "MOZ5", "MOZA", "MOZB", "MOZD", "MOZE", "MOZI", "MOZM", "MOZO", "MOZT",
-    "MOZW", "MOZSL01", "MOZSL02", "MOZSL03", "monline_dg", "monline_3_dg", "monline_4_dg",
-    "monline_7_dg", "firefox-a", "firefox-b", "firefox-b-1", "firefox-b-ab", "firefox-b-1-ab",
-    "firefox-b-d", "firefox-b-1-d", "firefox-b-e", "firefox-b-1-e", "firefox-b-m",
-    "firefox-b-1-m", "firefox-b-o", "firefox-b-1-o", "firefox-b-lm", "firefox-b-1-lm",
-    "firefox-b-lg", "firefox-b-huawei-h1611", "firefox-b-is-oem1", "firefox-b-oem1",
-    "firefox-b-oem2", "firefox-b-tinno", "firefox-b-pn-wt", "firefox-b-pn-wt-us", "ubuntu",
-    "ffab", "ffcm", "ffhp", "ffip", "ffit", "ffnt", "ffocus", "ffsb", "fpas",
-    "newext", "1000969a", null,
-)
 private val validChannelSet = setOf("ts")
 
 /**
  * Get a String in a specific format allowing to identify how an ads/search provider was used.
  *
  * @see [TrackKeyInfo.createTrackKey]
  */
+@Suppress("NestedBlockDepth", "ComplexMethod")
```

Comment:
In general I think both of these lint warnings are ones we want to avoid ignoring as much as possible. Would you file a follow-up so we can revisit the complexity of this function at a later date?

Reply:
Filed : https://bugzilla.mozilla.org/show_bug.cgi?id=1868381